app/src/main/java/com/blink22/android/mvpandroid/newTodo/NewTodoPresenter.java

```diff
@@ -0,0 +1,75 @@
+package com.blink22.android.mvpandroid.newTodo;
+
+import android.arch.lifecycle.Lifecycle;
+import android.arch.lifecycle.OnLifecycleEvent;
+import android.util.Log;
+
+import com.blink22.android.mvpandroid.models.Todo;
+import com.blink22.android.mvpandroid.network.TodosSubscriber;
+import com.blink22.android.mvpandroid.todos.TodosContract;
+import com.blink22.android.mvpandroid.todos.TodosPresenter;
+
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.observers.DisposableObserver;
+import rx.Subscription;
+import rx.subscriptions.CompositeSubscription;
+
+/**
+ * Created by ahmedghazy on 7/26/18.
+ */
+
+public class NewTodoPresenter implements NewTodoContract.Presenter {
+    private static final String TAG = TodosPresenter.class.getSimpleName();
+    private static NewTodoPresenter sNewTodoPresenter;
+
+    private final TodosSubscriber mTodosSubscriber;
+    private final NewTodoContract.View mView;
+    private CompositeDisposable mCompositeDisposable;
+
+    public static NewTodoPresenter getInstance(TodosSubscriber todosSubscriber, NewTodoContract.View view) {
+        return sNewTodoPresenter == null ? new NewTodoPresenter(todosSubscriber, view) : sNewTodoPresenter;
```

Comment:
@AFGhazy I believe this line needs a little modification, as `sNewTodoPresenter` will be null forever. Please let me know if I missed something.

It could be fixed as shown below. but I vote for implementing it as [here](https://github.com/AFGhazy/mvpAndroid/pull/1/files#diff-26f8d2899de602db3b0d11b38da76be2R32).
```
return sNewTodoPresenter == null ? sNewTodoPresenter = new NewTodoPresenter(todosSubscriber, view) : sNewTodoPresenter;
```
 Kindly let me know what you think.